jubilant/_juju.py

```diff
@@ -642,6 +642,43 @@ def run(
             if params_file is not None:
                 os.remove(params_file.name)
 
+    def ssh(
+        self,
+        target: str | int,
+        *command: str,
+        container: str | None = None,
+        host_key_checks: bool = True,
+        ssh_options: Iterable[str] = (),
+        user: str | None = None,
+    ) -> str:
+        """Executes a command using SSH on a machine or container and returns its standard output.
+
+        Args:
+            target: Where to run the command; this is a unit name such as ``mysql/0`` or a machine
+                ID such as ``0``.
+            command: Command to run, along with its arguments.
+            container: Name of container for Kubernetes charms. Defaults to the charm container.
+            host_key_checks: Set to False to disable host key checking (insecure).
+            ssh_options: OpenSSH client options, for example ``['-i', '/path/to/private.key']``.
+            user: User account to make connection with. Defaults to ``ubuntu`` account.
+        """
+        if not command:
+            raise TypeError('must provide a command')
+
+        args = ['ssh']
+        if container is not None:
+            args.extend(['--container', container])
+        if not host_key_checks:
+            args.append('--no-host-key-checks')
+        if user is not None:
+            args.append(f'{user}@{target}')
+        else:
+            args.append(str(target))
+        args.extend(ssh_options)
```

Comment:
I just realised that because `str` is also an `Iterable[str]`, this could silently have weird behaviour. Maybe we want:
```suggestion
        if isinstance(ssh_options, str):
            args.append(ssh_options)
        else:
            args.extend(ssh_options)
```

Reply:
Ah, good point. However, I'd prefer to raise TypeError as this is `ssh_options` plural. We can always loosen that restriction later if we want.